Framework/DataHandling/src/AsciiPointBase.cpp

```diff
@@ -59,7 +59,7 @@ void AsciiPointBase::exec() {
  */
 std::vector<double> AsciiPointBase::header(std::ofstream &file) {
   auto title = '#' + m_ws->getTitle();
-  const std::vector<double> &xTemp = m_ws->readX(0);
+  const auto &xTemp = m_ws->x(0);
   m_xlength = xTemp.size() - 1;
   std::vector<double> XData(m_xlength, 0);
   for (size_t i = 0; i < m_xlength; ++i) {
```

Comment:
`auto XData = m_ws->points(0)` should do the job, but the type is `Histogram::Points`. Have you considered changing the return type of this function?